### ISSUE TITLE
[Feature Request] Elasticsearch Re-Index

### ISSUE THREAD

**Issue by aprilmintacpineda** [aws-amplify/amplify-cli#3340]
**Describe the bug**

Elastic search's doc has a different schema compared to the one in `schema.graphql`. So when you use `@searchable` and do a `search query`, it will not do the intended behavior.

**Amplify CLI Version**

4.13.1

**To Reproduce**

With DynamoDB and graphql, create a schema that looks like so:

```
type Time
  @model
  @searchable
  @auth(rules: [{ allow: public, provider: apiKey }, { allow: private }]) {
  id: ID!
  totalHours: String
  timezone: String!
  week: String
  day: String
  createdAt: AWSDateTime
  updatedAt: AWSDateTime
}
```

**Notice**

- totalHours, day, and week are string and are intended to be strings.

look at the corresponding elastic search service that was created for this, the indices might still be blank and you will have to manually insert something on the dynamo db ([related issue](https://github.com/aws-amplify/amplify-cli/issues/2639)).

Insert the following data:

```js
{
  totalHours: '1.5'
  timezone: 'Asia/Manila'
  week: '2020-02-02'
  day: '3'
}
```

After the indices have been generated, check the index for the model above, and see the `doc` and you will see that totalHours has become `FLOAT` and week has become `DATE`.

WEIRD PART: `day` is a `TEXT`.

Now if you do a search query like this:

```graphql
{
  searchTimes (filter: {
    week: { eq: '2020-02-02' }
  }) {
     # put fields here
  }
}
```

This will return an empty array. You could use `matchPhrasePrefix` and it will return the correct data.

**Expected behavior**

Should follow `graphql.schema`.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Mac
 - v13.7.0

**Additional context**

Work around: prefix it with something like `test_${actualValue}`, haven't tried but might work if elastic search service is detecting the datatype by using the first input. This work around would only work if the index hasn't been created yet in elasticsearch, if the table has been indexed already, you might need to [delete that index](https://github.com/aws-amplify/amplify-cli/issues/3341) for it to be recreated.

**Comment by SwaySway** [comment]
Hi @aprilmintacpineda is there a reason you need the types themselves to be strings as opposed to the using float or date? 

The reason this is happening is because the lambda trigger will push the record from the dynamodb stream into elasticsearch. During this time the types being used in the record are inferred.

**Comment by aprilmintacpineda** [comment]
Hi @SwaySway :)

> The reason this is happening is because the lambda trigger will push the record from the dynamodb stream into elasticsearch. During this time the types being used in the record are inferred.

Oooooh, that's why, is this also why I experience [@searchable throws Elasticsearch:User IllegalArgumentException](https://github.com/aws-amplify/amplify-cli/issues/2639)?  So it really have to wait for that first record where the data type would be inferred from?

> is there a reason you need the types themselves to be strings as opposed to the using float or date?

The reason for the float being a string is because they are being displayed on the frontend as `1.00` or `1.50` or `1.55`, so if I save that as `float` then `1.00` might be saved as `1`, therefore I would have to parse it on the frontend to ensure that it shows `1.00`, I thought it's unnecessary to add that complexity on the frontend where I have to loop through each of those record when I could just save it as is because where's not really using it for anything else other than querying and displaying.

Now in the case of the date being a string, well we don't really need any other information other than that format itself, so again, I thought I could save us from adding unnecessary complexity by just saving it as is and we also don't use it for anything else other than querying and displaying.

**Comment by SwaySway** [comment]
Hi @aprilmintacpineda match and matchPhrasePrefix don't work with date types. I've opened pr to  clarify this a bit better. As for the issue with with the types not matching what is on the schema I'll refer to #3341 when reviewing indexing with the team.

**Comment by edwardfoyle** [comment]
The fix for this has been released in CLI v 4.32.0

**Comment by sacrampton** [comment]
Does this mean you are not inferring datatypes anymore? Inferring rather than specifically declaring is a significant cause of issues
